src/components/TeamBuilderPage/TeamBuilderPage.jsx

```diff
@@ -0,0 +1,179 @@
+                  synergies[alliance.key].level = index + 1;
+                  synergies[alliance.key].levelUnitCount = level.unitcount;
+                }
+              })       
+            }
+        })
+      })
+    })
+    this.setState({synergies}, () => {
+      ReactTooltip.rebuild();
+    });
+  }
+
+  handleSearchChange = (e) => {
+    this.setState({searchValue: e.target.value})
+  }
+
+  handleHeroSelection = h => () => {
+    if(this.state.team.length < 11) {
+      this.setState({team: [...this.state.team, h.key]}, this.computeAlliances)
+    }
+    this.resetSearch();
+  }
+
+  handleHeroDeSelection = i => () => {
+    const {team} = this.state;
+    team.splice(i, 1);
+    this.setState({team}, this.computeAlliances)
+  }
+
+  resetSearch = () => {
+    this.setState({searchValue: ""});
+  }
+
+  render() {
+    return(
+      <div className={commonStyles.PageContainer}>
+        <div className={styles.teamBuilderPageContainer}>
+          <div className={styles.heroesSelectionArea}>
+            <h2 style={{marginLeft: 4, marginBottom: 16}} className={styles.title}>{strings.select}</h2>
+          <div className={styles.searchInputContainer}>
+            <div className={styles.searchInput}>
+              <span className={styles.searchIcon}>&#8981;</span>
+              <input type="text" placeholder={strings.search_hero} value={this.state.searchValue} onChange={this.handleSearchChange}>
+              </input>
+            </div>
+          </div>
+            {
+              Object.keys(heroes)
+              .map(h => heroes[h])
+              .filter( h => new RegExp(this.state.searchValue, 'i').test(transformName(h.key)) && 
+                h.dota_unit_name.startsWith("npc_dota_hero_"))
+              .map(h =>
+                <img 
+                key={h.key} 
+                className={styles.heroImageSelectionArea} 
+                alt={h.displayName} 
+                src={GetHeroImage(h.dota_unit_name)}
+                onClick={this.handleHeroSelection(h)}
+                />
+              )             
+            }
+          </div>
+          <div className={styles.teamArea}>
+            <h2 className={styles.title}>{strings.team}</h2>
+              <div className={styles.selectedHeroesArea}>
+                {
+                  this.state.team.map((h, i) => {
+                    const hero = heroes[h];
+                    return(
+                      <div className={styles.heroImageTeamAreaContainer}>
+                          <img 
+                          key={hero.key + i}               
+                          alt={hero.displayName}
+                          className={styles.heroImageTeamArea} 
+                          src={GetHeroImage(hero.dota_unit_name)}
+                          onClick={this.handleHeroDeSelection(i)}
+                          />
+                          <div className={styles.heroName}>{transformName(h)}</div>
```

Comment:
Get the name from the localization file: https://github.com/odota/underlords-web/blob/master/src/components/HeroCard/HeroCard.tsx#L59